src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -5137,6 +5232,19 @@ theorem sub_le_sub_iff_le {x y z : BitVec w} (hxz : z ≤ x) (hyz : z ≤ y) :
     BitVec.toNat_sub_of_le (by rw [BitVec.le_def]; omega)]
   omega
 
+theorem sdivOverflow_eq_negOverflow_of_allOnes {w : Nat} {x y : BitVec w} (hy : y = allOnes w) :
+    sdivOverflow x y = negOverflow x := by
+  simp only [sdivOverflow, hy, toInt_allOnes, Int.reduceNeg, ge_iff_le, negOverflow]
+  rcases w with _|w
+  · simp [of_length_zero]
+  · simp only [Nat.add_one_sub_one, Nat.zero_lt_succ, ↓reduceIte, Int.reduceNeg, Int.ediv_neg,
+      Int.ediv_one, Int.neg_lt_neg_iff]
+    have xle := le_two_mul_toInt (x := x); have xlt := two_mul_toInt_lt (x := x)
+    simp only [bool_to_prop]
+    by_cases hx : x.toInt = - 2 ^ w
+    · simp [hx]
+    · simp [show ¬x.toInt == -2 ^ w by simp only [beq_iff_eq, hx, not_false_eq_true]]; omega
```

Comment:
I'm a bit surprised that this proof is so long! In my naive imagination, we should that `x / allOnes w = -x`, and the result immediately follows that sdiv overflowing is equivalent to negation overflowing in this case. 

If this proof strategy does not work, a comment clarifying why this proof strategy does not work would be appreiated!

Reply:
added 
`theorem sdiv_allOnes {w : Nat} {x : BitVec w} : 
    x.toInt / (allOnes w).toInt = - x.toInt`
to make the strategy cleared